### ISSUE TITLE
queryString producing an empty statement for Batch

### ISSUE THREAD

**Issue by leoromanovsky** [outworkers/phantom#687]
Version: 2.9.0

Calling `queryString` on a batch produces `;`

Sample code:

```
println("insert", insertStatement.queryString)
val query = Batch.unlogged.add(insertStatement).add(insertStatement)
println("query", query.iterator.size, query.queryString)
```

Result:

```
(insert,INSERT INTO test.my_table (col1, col2, col3, col4) VALUES(1, 2, 3, 4);)
(query,2,;)
```

I upgraded to 2.9.0 after seeing a related issue (https://github.com/outworkers/phantom/issues/647) - do you think something else is needed to fix this?

**Comment by alexflav23** [comment]
Hi @leoromanovsky,

We have an internal ticket to add this, however I feel obliged to tell you it does require extra work to produce this. Please accept my apologies if I am in any way stating the obvious, but a `BoundStatement` will have an `id` that makes the cluster aware what the query to execute is.

When you call `bind`, a macro is invoked, to automatically generate the necessary code to deserialize concrete types to bytes, e.g`ByteBuffer`. If you want to produce a queryString, you need to drop some of that performance and call to string, and retrace that in the original query.

I'm making you aware you are paying twice the cost of serialisation, which may not be trivial in all cases. That aside, we can definitely implement an accurate `queryString`.

**Comment by alexflav23** [comment]
Hi @leoromanovsky,

Ok this will be fixed in the next release, but with a catch. To trigger automated creation of queryStrings for batch statements, you need to:

```scala
import com.outworkers.phantom.macros.debug.options.ShowBoundStatements
```

This will internally change the behaviour of the macros that generate the serializers, so when you want performance or in the default case, we do not alter the behaviour users rely on. The macros are summoned at the point where you call `.bind`, so that's where you will need to have this import available in scope.

Hope this makes sense and gives you what you need!

Regards,
Flavian

**Comment by alexflav23** [comment]
Hi @leoromanovsky, just waiting on CI now, 2.9.1 will be auto-published shortly. Hope it works for you!

**Comment by leoromanovsky** [comment]
Thanks Flavian!

**Comment by alexflav23** [comment]
Hi @leoromanovsky I think I may have confused tickets when reporting progress here, please re-confirm if I have, you were talking about `Batch` I solved the problem for `BoundStatement` :). Apologies, let us know and it will be picked up again!

**Comment by alexflav23** [comment]
Hi @leoromanovsky,

Fixed for the next release, batches now automatically print their query strings. Blocked on #695, but we'll do a separate release if that drags beyond Friday.

Regards.

**Comment by alexflav23** [comment]
Hi @leoromanovsky Fixed in 2.10.0, which also fixes runtime errors calling the `store` method through some implicit macro machinery.
